src/DynamoCoreWpf/Views/Input/ParameterEditor.xaml.cs

```diff
@@ -0,0 +1,249 @@
+        {
+            Color color = (Color)ColorConverter.ConvertFromString("#2E998F");
+            var classHighlightRule = new HighlightingRule();
+            classHighlightRule.Color = new HighlightingColor()
+            {
+                Foreground = new Dynamo.Wpf.Views.CodeBlockEditorUtils.CustomizedBrush(color)
+            };
+
+            var engineController = dynamoViewModel.Model.EngineController;
+            var wordList = engineController.CodeCompletionServices.GetClasses();
+            String regex = String.Format(@"\b({0})({0})?\b", String.Join("|", wordList));
+            classHighlightRule.Regex = new Regex(regex);
+
+            return classHighlightRule;
+        }
+
+        #endregion
+
+        #region Event handlers
+
+        private void OnTextAreaTextEntering(object sender, TextCompositionEventArgs e)
+        {
+            if (e.Text.Length > 0)
+            {
+                char currentChar = e.Text[0];
+                if (completionWindow == null)
+                {
+                    string validCharacters = "[]=.:\"";
+                    if (!char.IsWhiteSpace(currentChar) &&
+                        !char.IsLetterOrDigit(currentChar) && 
+                        !validCharacters.Contains(currentChar))
+                    {
+                        e.Handled = true;
+                    }
+                }
+                else
+                {
+                    if (currentChar == '\n' || currentChar == '\r')
+                    {
+                        completionWindow.CompletionList.RequestInsertion(e);
+                    }
+                    else if (!char.IsLetterOrDigit(currentChar))
+                    {
+                        completionWindow.Close();
+                    }
+                }
+            }
+        }
+
+        private void OnTextAreaTextEntered(object sender, TextCompositionEventArgs e)
+        {
+            if (!InnerTextEditor.Text.Contains(':'))
+                return;
+
+            string type = InnerTextEditor.Text.Split(':').Last().Trim();
+            var types = this.GetMatchedTypes(type);
+            if (types == null || !types.Any())
+                return;
+
+            int index = InnerTextEditor.Text.IndexOf(':');
+            if (index < 0 || InnerTextEditor.CaretOffset <= index)
+                return;
+
+            ShowCompletionWindow(types, InnerTextEditor.Text.Length - type.Length);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void OnTextAreaLostFocus(object sender, RoutedEventArgs e)
+        {
+            InnerTextEditor.TextArea.ClearSelection();
+
+            nodeViewModel.DynamoViewModel.ExecuteCommand(
+                new DynCmd.UpdateModelValueCommand(
+                    nodeViewModel.NodeModel.GUID, "InputSymbol",
+                    InnerTextEditor.Text));
```

Comment:
Please do not send the command if the text contents are identical (e.g. losing focus without actual changes). That has caused issues with undo [which was subsequently fixed](https://github.com/DynamoDS/Dynamo/pull/3532/files#diff-05003f9e6533825cf76ac9a0a86e5109R432), do look at the changes for details.